src/plugins/install/index.rst

```diff
@@ -1,4 +1,157 @@
+   :template: docstring_only.rst
+
+   cylc.flow.install_plugins.log_vc_info
+
+.. Note: Autosummary generates files in this directory, these are cleaned
+         up by `make clean`.
+
+Developing ``pre_configure`` and ``post_install`` plugins
+---------------------------------------------------------
+
+Cylc uses entry points registered by setuptools to search for pre-configure
+and post-install plugins.
+
+Hello World
+^^^^^^^^^^^
+
+In this example a pre-configure plugin which logs a "Hello World" message
+and, after installation, logs some info about the installation:
+
+.. code-block:: python
+   :caption: my_plugin.py
+
+   from cylc.flow import LOG
+
+   def pre_configure(srcdir=None, opts=None, rundir=None):
+      # write Hello to the Cylc log.
+      LOG.info(f'Hello World')
+      return {}
+
+   def post_install(srcdir=None, opts=None, rundir=None):
+      LOG.info(f'installed from {srcdir}')
+      LOG.info(f'installed to {rundir}')
+      LOG.info(f'installation options were {options}')
+      return {}
+
+Plugins are registered by registering them with the ``cylc.pre_configure``
+and ``cylc.post_install`` entry points:
+
+.. code-block:: python
+   :caption: setup.py
+
+   # plugins must be properly installed, in-place PYTHONPATH meddling will
+   # not work.
+
+   from setuptools import setup
+
+   setup(
+       name='my-plugin',
+       version='1.0',
+       py_modules=['my_plugin'],
+       entry_points={
+          # register this plugin with Cylc
+          'cylc.pre_configure': [
+            # name = python.namespace.of.module
+            'my_plugin=my_plugin.my_plugin:pre_configure'
+          ]
+          'cylc.post_install': [
+            # name = python.namespace.of.module
+            'my_plugin=my_plugin.my_plugin:post_install'
+          ]
+       }
+   )
+
+API Reference
+^^^^^^^^^^^^^
+
+Cylc will pass following arguments to pre-configure and post-install plugins:
+
+``srcdir`` (Path or string)
+   The directory from which ``cylc install`` is installing a workflow,
+   or the directory passed to ``cylc validate``, ``cylc graph`` and other
+   CLI commands which work without installing a workflow.
+``options`` (Namespace)
+   CLI options set for a Cylc script.
+``rundir`` (Path or string)
+   The destination of a ``cylc install`` or ``reinstall`` command.
+
+The pre-confugure plugin should return a dictionary which may contain the
+following keys:
+
+``env``
+   A dictionary of environment variables to be exported to the scheduler
+   environment.
+``template_variables``
+   A dictionary of template variables to be used by Jinja2 or Empy when
+   templating the workflow configuration files.
+``templating_detected``
+   ``jinja2`` | ``empy`` to be used when templating. N.b: This will result in
+   failure if the templating language set does not match the shebang line of
+   the ``flow.cylc`` file.
+
+The post-install entry point does not return any data used by Cylc.
+
+More advanced example
```

Comment:
Perhaps say, "see cylc-rose" for a concrete implementation.

Reply:
Instead of this section, or as well as?